mp4parse/src/lib.rs

```diff
@@ -1419,10 +1424,15 @@ impl AvifContext {
                     }
                 }
                 unreachable!(
-                    "IsobmffItem::Location requires the location exists in AvifContext::media_storage"
+                    "IsobmffItem::MdatLocation requires the location exists in AvifContext::media_storage"
                 );
             }
-            IsobmffItem::IdatLocation(_) => unimplemented!(),
+            IsobmffItem::IdatLocation(extent) => {
+                self.item_data_box
+                    .as_ref()
+                    .and_then(|idat| idat.get(extent))
+                    .unwrap_or_else(|| unreachable!("IsobmffItem::IdatLocation equires the location exists in AvifContext::item_data_box"))
```

Comment:
Typo in "requires".

nit: I wonder if these uses of `unreachable!` should use `panic!` instead, since they're potentially parsing errors rather than code paths we never expect to hit.

Reply:
I definitely agree with you that it _should_ be `panic!` if there's any way to reach this code, even with bad input. My assertion here (and in the above, mdat case) by using `unreachable!` is that there _isn't_ an input which will be able to reach that, and if it _is_ hit, that implies an error in the logic itself (and possibly bad input as well), not just bad input. In the case that I just haven't finished something, I use `unimplemented!`, but that's not the case here any longer.

In this specific case, I believe the only way there can be a `IsobmffItem::IdatLocation(extent)` value stored in `AvifItem::image_data`, is in the case that `AvifContext::item_data_box` has already been populated with a `DataBox` that covers `extent`. If my reasoning here is wrong, we should change it.

Reply:
Sorry, I should've looked more closely before making that suggestion - agree it's used appropriate in these situations.